ui/src/components/Table/DataTable.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  ColumnDef,
+  SortingState,
+  VisibilityState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { useState } from "react";
+
+import {
+  DropdownMenu,
+  DropdownMenuCheckboxItem,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+  Button,
+  Input,
+} from "@flow/components";
+import { useT } from "@flow/providers";
+
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from ".";
+
+interface DataTableProps<TData, TValue> {
+  columns: ColumnDef<TData, TValue>[];
+  data: TData[];
+  selectColumns?: boolean;
+  showFiltering?: boolean;
+}
+
+function DataTable<TData, TValue>({
+  columns,
+  data,
+  selectColumns = false,
+  showFiltering = false,
+}: DataTableProps<TData, TValue>) {
+  const t = useT();
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({});
+  const [rowSelection, setRowSelection] = useState({});
+  const [globalFilter, setGlobalFilter] = useState("");
```

Comment:
If any of these are being managed already by react-table, and accessible through useReactTable, I don't think we need to keep state here as well

Reply:
I followed the guide from here: https://ui.shadcn.com/docs/components/data-table

As I understand this, react-table does expose all the functionality but the state still has to be managed at the component. This can be quite generic in our case. 
For example - columnVisibilty we can just the `selectColumns` to true and we can use the button to select which columns to show. The alternative is we pass the `[columnVisibility, setColumnVisibility]` state into the table. 

Opinion:
Keeping it simple for now should be okay, can extend if needed once we get better understanding on how the user will use this and what kind of data will be there.